awx/main/models/schedules.py

```diff
@@ -103,70 +103,54 @@ class Meta:
-                  'TZID': dateutil.tz.gettz('America/New_York')
-                }
-           })`
+        ...and we pass a hint about the local timezone to dateutil's parser:
+        `dateutil.rrule.rrulestr(rrule, {
+            'tzinfos': {
+                'TZID': dateutil.tz.gettz('America/New_York')
+              }
+         })`
 
-          it's possible that we can remove the custom code that performs this
-          parsing if TZID= gains support in upstream dateutil:
-          https://github.com/dateutil/dateutil/pull/615
-
-        * RFC5545 specifies that: if the "DTSTART" property is specified as
-          a date with local time (in our case, TZID=), then the UNTIL rule part
-          MUST also be treated as a date with local time.  If the "DTSTART"
-          property is specified as a date with UTC time (with a Z suffix),
-          then the UNTIL rule part MUST be specified as a date with UTC time.
-
-          this function provides additional parsing to translate RRULES like this:
-
-          DTSTART;TZID=America/New_York:20180601T120000 RRULE:FREQ=DAILY;INTERVAL=1;UNTIL=20180602T170000
-
-          ...into this (under the hood):
-
-          DTSTART;TZID=America/New_York:20180601T120000 RRULE:FREQ=DAILY;INTERVAL=1;UNTIL=20180602T210000Z
+        it's likely that we can remove the custom code that performs this
+        parsing if TZID= gains support in upstream dateutil:
+        https://github.com/dateutil/dateutil/pull/619
         """
-        source_rrule = rrule
+        kwargs['forceset'] = True
         kwargs['tzinfos'] = {}
         match = cls.TZID_REGEX.match(rrule)
         if match is not None:
             rrule = cls.TZID_REGEX.sub("DTSTART\g<stamp>TZI\g<rrule>", rrule)
             timezone = gettz(match.group('tzid'))
-            if 'until' in rrule.lower():
-                # if DTSTART;TZID= is used, coerce "naive" UNTIL values
-                # to the proper UTC date
-                match_until = re.match(".*?(?P<until>UNTIL\=[0-9]+T[0-9]+)(?P<utcflag>Z?)", rrule)
-                until_date = match_until.group('until').split("=")[1]
-                if len(match_until.group('utcflag')):
-                    raise ValueError(six.text_type(
-                        _('invalid rrule `{}` includes TZINFO= stanza and UTC-based UNTIL clause').format(source_rrule)
-                    ))  # noqa
-                localized = make_aware(
-                    datetime.datetime.strptime(until_date, "%Y%m%dT%H%M%S"),
-                    timezone
-                )
-                utc = localized.astimezone(pytz.utc).strftime('%Y%m%dT%H%M%SZ')
-                rrule = rrule.replace(match_until.group('until'), 'UNTIL={}'.format(utc))
             kwargs['tzinfos']['TZI'] = timezone
         x = dateutil.rrule.rrulestr(rrule, **kwargs)
 
+        for r in x._rrule:
+            if r._dtstart and r._until:
+                if all((
+                    r._dtstart.tzinfo != dateutil.tz.tzlocal(),
```

Comment:
If `DTSTART` specifies a timezone (including UTC via `Z`)
_and_
`UNTIL` is *not* in UTC...